src/application/service/index.ts

```diff
@@ -0,0 +1,20 @@
+import { GenerateAuthToken } from './GenerateAuthToken';
+import { ExchangeAuthCode } from './ExchangeAuthCode';
+import { CreateNewMagicLink } from './CreateNewMagicLink';
+import * as config from '../../config';
+import { GetExistingOrCreateNewUser } from './GetExistingOrCreateNewUser';
+import { magicLinkRepository } from '../../infrastructure/persistence';
```

Comment:
Can we export default in all the classes to avoid this kind of syntax? Also, what's the benefit of using an index here instead of importing the files directly when needed? This file looks ugly and very manual. Also, the way you are initing things here works for now because everything is known at interpret-time... but that will probably not be true very soon, so we'll have to move the dependency injection to the controllers

Reply:
don't agree with this point. The point of this file is to make it easy for when we introduce dependency injection. Now at least the wirings are in one file (even if the arrows are in the wrong direction) and it's faster than researching how to do DI in node :P Let's cross that bridge when we get there

Reply:
On the default, the main reason I am using this over default is that it is easier for the IDEs (At least intelliJ) to discover what variables to import (and refactor) because unlike default they are named. Not a strong argument.

Reply:
Ok, I'm fine with the IDE argument. About the other one, I think you didn't get my point. If you need to inject something not known at the interpreter time (which is pretty normal), this way will simply not work.

Anyway, let's start simple and leave it like this until we need something more complex, ok